plugins/Profiler/views/SnapshotFlamegraph.js

```diff
@@ -257,16 +264,18 @@ class ListItem extends PureComponent<any, void> {
           }
 
           const actualDuration = fiber.get('actualDuration') || 0;
+          const selfBaseDuration = fiber.get('selfBaseDuration') || 0;
           const name = fiber.get('name') || 'Unknown';
           const didRender = committedNodes.includes(id);
+          const color = didRender ? colorBySelfTime ? getGradientColor(selfBaseDuration / snapshot.maxSelfBaseDuration) : getGradientColor(actualDuration / maxDuration) : didNotRender;
 
           return (
             <ChartNode
-              color={didRender ? getGradientColor(actualDuration / maxDuration) : didNotRender}
+              color={color}
               height={barHeight}
               isDimmed={index < focusedNodeIndex}
               key={id}
-              label={didRender ? `${name} (${actualDuration.toFixed(1)}ms)` : name}
+              label={didRender ? `${name} (${colorBySelfTime ? selfBaseDuration.toFixed(1) : actualDuration.toFixed(1)}ms)` : name}
```

Comment:
I think I've changed my mind on the label after playing with it for a few minutes. For the self-time case, I think we should actually show "self time / tree time". Otherwise it feels confusing to me to see smaller values above larger values.

Reply:
Showing `selfBaseDuration` / `actualDuration` makes the minor discrepancies I mentioned too obvious. I think this confirms that we should just calculate a "self duration" value.

Reply:
I'm going to play with this a little locally and get back to you!